### PR TITLE
fix: added dependency in basic example

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

An error was occurring when trying to delete the read-only database saying that the main database hasn't been deleted yet. A dependency was added to stop it from trying to delete before the main database finished deleting.

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

A bug was fixed.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -40,4 +40,5 @@ module "read_only_replica_postgresql_db" {
   remote_leader_crn = module.postgresql_db.crn
   member_memory_mb  = 2304  # Must be an increment of 384 megabytes. The minimum size of a read-only replica is 2 GB RAM
   member_disk_mb    = 15360 # Must be an increment of 512 megabytes. The minimum size of a read-only replica is 15.36 GB of disk
+  depends_on        = [module.postgresql_db]
```
Comment (ocofaigh) [inline examples/basic/main.tf:43]: using explicit `depends_on` like this is not recommended by terraform unless there is no other option (see nice [article](https://itnext.io/beware-of-depends-on-for-modules-it-might-bite-you-da4741caac70) on this). In this case there is already an implicit dependency in place due to the fact this line: `remote_leader_crn = module.postgresql_db.crn` so no need to add an explicit one. 

For the problem you are trying to solve, please re-check the details I added to https://github.ibm.com/GoldenEye/issues/issues/5484 on what the solution had to be here. We need to either add a hard sleep, or call an api to check status before attempting to delete the read only replica. This is what the ICD team recommended in the support case I had opened

**Comment by kierramarie** [comment]
I know depends_on are not good to use however I was following the [documentation](https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep) for time_sleep. I tried a different way that doesn't use depends_on but it didn't work.

**Comment by ocofaigh** [comment]
@kierramarie Yes if you are going to use a time sleep for this workaround, then you will need to use `depends_on`, but I didnt see any time sleep in this PR?

File: examples/basic/main.tf [hunk 2]
```diff
@@ -24,6 +24,12 @@ module "postgresql_db" {
   access_tags       = var.access_tags
 }
 
+resource "time_sleep" "wait_time" {
```
Comment (kierramarie) [inline examples/basic/main.tf:27]: @ocofaigh I added the time_sleep here

Comment (ocofaigh) [inline examples/basic/main.tf:27]: Can you leave a comment above this to say:

> On destroy, we are seeing that even though the replica has been returned as destroyed by terraform, the leader instance destroy can fail with: "You must delete all replicas before disabling the leader. Try again with valid values or contact support if the issue persists."
> The ICD team have recommended to wait for a period of time after the replica destroy completes before attempting to destroy the leader instance, so hence adding a time sleep here.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**